core/templates/pages/admin-page/features-tab/admin-features-tab.component.spec.ts

```diff
@@ -130,37 +135,52 @@ describe('Admin page feature tab', function() {
 
       component.addNewRuleToBottom(featureFlag);
       expect(featureFlag.rules.length).toBe(2);
-      expect(featureFlag.rules[1].valueWhenMatched).toBeFalse();
+      expect(featureFlag.rules[0].valueWhenMatched).toEqual('original');
     });
   });
 
   describe('.removeRule', () => {
     it('should remove rule', () => {
       const featureFlag = component.featureFlags[0];
+      component.addNewRuleToBottom(featureFlag);
+      featureFlag.rules[1].valueWhenMatched = '1';
 
-      expect(featureFlag.rules.length).toBe(1);
       component.removeRule(featureFlag, 0);
-      expect(featureFlag.rules.length).toBe(0);
+
+      expect(featureFlag.rules.length).toBe(1);
+      expect(featureFlag.rules[0].valueWhenMatched).toEqual('1');
     });
   });
 
   describe('.moveRuleUp', () => {
     it('should move rule up', () => {
       const featureFlag = component.featureFlags[0];
       component.addNewRuleToBottom(featureFlag);
+      featureFlag.rules[1].valueWhenMatched = '1';
+      component.addNewRuleToBottom(featureFlag);
+      featureFlag.rules[2].valueWhenMatched = '2';
 
       component.moveRuleUp(featureFlag, 1);
-      expect(featureFlag.rules[0].valueWhenMatched).toBeFalse();
+
+      expect(featureFlag.rules[0].valueWhenMatched).toEqual('1');
```

Comment:
Maybe add a clarifying comment, e.g.:

```python
# Verify rule order 'original' -> '1' -> '2' now reflects the moved rule '1'.
```

Ditto elsewhere for tests where the asserted behavior may not be immediately obvious when glancing at the test.